internal/database/database.go

```diff
@@ -22,25 +22,10 @@ import (
 const (
 	billingContainer       = "Billing"
 	locksContainer         = "Locks"
-	operationsContainer    = "Operations"
 	resourcesContainer     = "Resources"
-	subscriptionsContainer = "Subscriptions"
 	partitionKeysContainer = "PartitionKeys"
 
-	// XXX The azcosmos SDK currently only supports single-partition queries,
-	//     so there's no way to list all items in a container unless you know
-	//     all the partition keys. The backend needs to list all items in the
-	//     Operations container so to work around this limitation we keep all
-	//     items in a single partition with a well-known name: "workaround".
-	//
-	//     Once [1] is fixed we could transition the Operations container to
-	//     using subscription IDs as the partition key like other containers.
-	//     The items are transient thanks to the container's default TTL, so
-	//     GetOperationDoc would just need temporary fallback logic to check
-	//     the "workaround" partition.
-	//
-	//     [1] https://github.com/Azure/azure-sdk-for-go/issues/18578
-	operationsPartitionKey = "workaround"
+	operationTimeToLive = 604800 // 7 days
```

Comment:
Q: is this arbitrary or an ARM requirement?

Reply:
[Copied from ARO Classic](https://github.com/Azure/ARO-RP/blob/826deb02558cc4380bf7b6a00f348752f2e42a36/pkg/deploy/assets/rp-production.json#L723).  I haven't found any firm requirements about this from Microsoft.

Reply:
Actually I had another look just now and found some new information by way of [Stack Overflow](https://stackoverflow.microsoft.com/questions/308952).

[Header Based Async Operation Timeout](https://eng.ms/docs/products/arm/rp_onboarding/headerbasedasyncoperationtimeout)

This is a newer approach.  The older approach in the era of ARO Classic was to specify an [async timeout in the resource manifest](https://eng.ms/docs/products/arm/rp_onboarding/resourceproviderregistration#asynctimeoutrules-optional).  The maximum allowed timeout is 6 days, so my guess is 7 days was chosen to add some buffer.  But the true rationale has probably been lost to staff attrition.

If we want to re-evaluate, this is probably @bennerv question for when he gets back.